src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -1207,8 +1208,7 @@ private void setupGCSClient(
         // We are authenticated with an oauth access token.
         this.gcsClient =
             StorageOptions.newBuilder()
-                .setHeaderProvider(
-                    FixedHeaderProvider.create("Authorization", "Bearer " + accessToken))
+                .setCredentials(GoogleCredentials.create(new AccessToken(accessToken, null)))
```

Comment:
Add this line based on a new connection option. Default value of that option you can discuss with support, depends on howmany cases of default credentials use. This way if some new behavior occurs for existing customer base, they can turn it on/off in the driver and fallback with previous behavior. GCS changes are very sensitive, so extra precausion will help us.

Reply:
I have added a new connection parameter `disableGcsDefaultCredentials` with default `false` to keep current behaviour unchanged